server/src/test/java/keywhiz/service/daos/SecretContentDAOMockTest.java

```diff
@@ -0,0 +1,119 @@
+package keywhiz.service.daos;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import keywhiz.KeywhizConfig;
+import keywhiz.jooq.tables.SecretsContent;
+import keywhiz.service.crypto.RowHmacGenerator;
+import org.apache.commons.lang3.StringUtils;
+import org.jooq.DSLContext;
+import org.jooq.Record1;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+import org.jooq.tools.jdbc.MockConnection;
+import org.jooq.tools.jdbc.MockDataProvider;
+import org.jooq.tools.jdbc.MockExecuteContext;
+import org.jooq.tools.jdbc.MockResult;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+
+public class SecretContentDAOMockTest {
```

Comment:
There's already a SecretContentDAOTest but it relies on dependency injection via a JUnit Rule to instantiate the object graph. The problem with this is that the configuration file is hardcoded so we have no clean way of using different configurations (we could grab the injected config and update the value, but the whole injection thing is a lot of overhead for something which should be easy to test).